### DISCUSSION TITLE
How to enable biometric option

### DISCUSSION THREAD

**Discussion by sharukhkhan234** [horilla-opensource/horilla#194] [General]
Hi! I am looking for biometric option in the left menu bar, but did not find it. please share complete details, how to can I enable biometric option

**Comment by horilla-opensource** [comment]
Hi @sharukhkhan234 ,
Can you confirm if the biometric app is added in the installed apps list?

**Reply by sharukhkhan234** [reply]
No, I did not add create an app for biometric devices, I was thinking that the biometric app would come as it like others are installed during installation because I saw that option in the documentation, so I was hoping that the app might be available, should I create a separate app for biometric device?

**Reply by horilla-opensource** [reply]
Hi @sharukhkhan234 ,
The biometric app was not added by default in the master code, so you'll have to add it explicitly it to the settings.
We are going to add the biometric app to the master code base today.
We'll let you know once we add that.

With Regards,
**Team Horilla**

**Reply by sharukhkhan234** [reply]
ok thank you! I am waiting.........

**Reply by sharukhkhan234** [reply]
Hi Team Horilla! 
is biometric device app added?

**Reply by horilla-opensource** [reply]
Hi @sharukhkhan234 ,
Yes it is updated to the master code base.
You can pull the master branch for the biometric app.

With Regards,
**Team Horilla**

**Reply by sharukhkhan234** [reply]
Do you think I should Reinstall it, or will it work?

**Reply by horilla-opensource** [reply]
If you are working with our test database, I suggest you do a fresh pull and start.

**Reply by sharukhkhan234** [reply]
It's okay!
It works fine now, Appreciated your work. 
We need a little bit of improvement regarding the biometric app when we do check-in on the device, it should not be check-in before checked-out, the conflict is that now we can do multiple check-in before check-out, an improvement is to avoid user from multiple check-in in a day, and users should check-in once and then check-out. because for testing we did multiple check-in today on the device and in the software shows multiple with the time.

Regards.

**Reply by sharukhkhan234** [reply]
And the second thing is need to be add in the biometric device app is "face" detection. for now it only works on fingers and we need to add face as well. It should be work on fingers which is working and "face". this is a kind request regarding these issue, please check if possible.

Regards,

**Reply by horilla-opensource** [reply]
Hi @sharukhkhan234 ,
The face id check in also works with the present system.
The settings will show only the configuration of which finger is configured in the machine, but it do recognize the face ids.
From the code we only take the details whether the signal is for check in or check out and mark the attendance accordingly. Whatever means be the check in (Face or Finger) the signal received from the machine is same and we make use of that. 
We have tested the face recognition with our ZK machine and it is working.

With Regards,
**Team Horilla**
